src/core/net/dnssd_server.hpp

```diff
@@ -107,11 +132,10 @@ class Server : public InstanceLocator, private NonCopyable
             kUnknownOffset = 0, // Unknown offset value (used when offset is not yet set).
         };
 
+        explicit NameCompressInfo(void) = default;
+
         explicit NameCompressInfo(const char *aDomainName)
```

Comment:
```suggestion
        explicit NameCompressInfo(const char *aDomainName=nullptr)
```

The members are not initialized in the default constructor.

Reply:
It's actually fine not to initialize it.